azure-devops/azext_devops/test/repos/test_ref.py

```diff
@@ -0,0 +1,82 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import unittest
+
+try:
+    # Attempt to load mock (works on Python 3.3 and above)
+    from unittest.mock import patch, ANY
+except ImportError:
+    # Attempt to load mock (works on Python version below 3.3)
+    from mock import patch, ANY
+
+from azext_devops.dev.common.services import clear_connection_cache
+from azext_devops.dev.repos.ref import (list_refs, create_ref, delete_ref, update_ref)
+
+
+class TestRefMethods(unittest.TestCase):
+
+    _TEST_DEVOPS_ORGANIZATION = 'https://AzureDevOpsCliTest.visualstudio.com'
```

Comment:
Put an invalid url here like.. https://some-invalid-org.visualstudio.com. 

This will fail the test since the "client = get_git_client(organization)" call in each of the commands makes a network call. You will need to mock it, in your case to return a GitClient object. 

We discovered this recently and I am fixing other tests. 

This one is implemented correctly. 
https://github.com/Microsoft/azure-devops-cli-extension/blob/master/azure-devops/azext_devops/test/pipelines/test_pipelines_build.py

check -  self.get_client = patch('vsts.vss_connection.VssConnection.get_client')